File: database-client/src/main/java/com/example/demo/Post.java
package com.example.demo;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import io.r2dbc.postgresql.codec.Json;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

/**
 * @author hantsy
 */
@Data
@ToString
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Post {

    private UUID id;

    private String title;

    private String content;

    @JsonSerialize(using = PgJsonObjectSerializer.class)
    @JsonDeserialize(using = PgJsonObjectDeserializer.class)
    @Builder.Default
    private Json metadata = Json.of("{}");

    @Builder.Default
    private Status status = Status.DRAFT;

    private LocalDateTime createdAt;

    enum Status {
        DRAFT, PENDING_MODERATION, PUBLISHED;
    }

}

File: komapper/src/main/kotlin/com/example/demo/Post.kt
package com.example.demo

import org.komapper.annotation.EnumType.NAME
import org.komapper.annotation.KomapperAutoIncrement
import org.komapper.annotation.KomapperColumn
import org.komapper.annotation.KomapperCreatedAt
import org.komapper.annotation.KomapperEntity
import org.komapper.annotation.KomapperEnum
import org.komapper.annotation.KomapperId
import org.komapper.annotation.KomapperTable
import org.komapper.annotation.KomapperUpdatedAt
import java.time.LocalDateTime

@KomapperEntity
@KomapperTable("posts")
data class Post(
    @KomapperId
    @KomapperAutoIncrement
    @KomapperColumn("id")
    val id: Long? = null,

    @KomapperColumn("title")
    val title: String,

    @KomapperColumn("body")
    val body: String,

    @KomapperEnum(type = NAME)
    val status: Status = Status.DRAFT,

    @KomapperCreatedAt
    @KomapperColumn("created_at")
    val createdAt: LocalDateTime? = null,

    @KomapperUpdatedAt
    @KomapperColumn("updated_at")
    val updatedAt: LocalDateTime? = null,
) {
    enum class Status {
        DRAFT, PUBLISHED
    }
}